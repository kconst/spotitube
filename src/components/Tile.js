import React, { PropTypes, Component } from 'react';

export default class Tile extends Component {
  render() {
    const { name, description, images } = this.props;

    return (
      <li className="Tile">
        <img src={ images[0] && images[0].url } alt={ name }/>
        <h2>
          { name }
        </h2>

        <p>
          { description }
        </p>
      </li>
    )
  }
}

Tile.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};