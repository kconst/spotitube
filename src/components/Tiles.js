import React, { PropTypes, Component } from 'react';
import Tile from './Tile';

export default class Tiles extends Component {
  render() {
    return (
      <ul>
      { (() => {
        if (this.props.data.length > 0) {
          return this.props.data.map((tile, i) =>
            <Tile
              select={ this.props.select }
              key={ i }
              { ...tile }
            />
          )
        } else {
          return <h4>No results were found from your query.</h4>;
        }
      })() }
  </ul>
    )
  }
}

Tiles.propTypes = {
  data: PropTypes.array.isRequired
};