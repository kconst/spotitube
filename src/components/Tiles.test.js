import React from 'react';
import Tiles from './Tiles';
import Tile from './Tile';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

const data = [
    {
      id: "3xIOCwSEW9WdW5auFWBJfw",
      images: [
        { url: "https://i.scdn.co/image/68282faa891fba544f90e32cfcee6e23ff2b6d1c" }
      ],
      name: "St. Lucia – Spotify Sessions"
    },
    {
      id: "5f9AduFWkiRl4XjNsiIvyh",
      images: [
        { url: "https://i.scdn.co/image/ed46b9b221b4e19da42c4adf8f4db6c649339fe9" }
      ],
      name: "Megadeth – ENDGAME"
    }
  ],
  jsx = <Tiles data={ data }></Tiles>;

test('snapshot comparison', () => {
  const component = renderer.create(jsx),
    tile = component.toJSON();
  
  // snapshot check
  expect(tile).toMatchSnapshot();
});

test('it should have multiple tiles', () => {
  const tiles = shallow(jsx);
  
  expect(tiles.find('Tile').length).toEqual(2);
});