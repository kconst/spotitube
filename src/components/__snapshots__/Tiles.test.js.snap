exports[`test snapshot comparison 1`] = `
<ul>
  <li
    className="Tile"
    onClick={[Function]}>
    <img
      alt="St. Lucia – Spotify Sessions"
      src="https://i.scdn.co/image/68282faa891fba544f90e32cfcee6e23ff2b6d1c" />
    <h2>
      St. Lucia – Spotify Sessions
    </h2>
  </li>
  <li
    className="Tile"
    onClick={[Function]}>
    <img
      alt="Megadeth – ENDGAME"
      src="https://i.scdn.co/image/ed46b9b221b4e19da42c4adf8f4db6c649339fe9" />
    <h2>
      Megadeth – ENDGAME
    </h2>
  </li>
</ul>
`;
